tests/StripeFormattingTrait.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace BigGive\Identity\Tests;
+
+use ArrayIterator;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Prophecy\Prophecy\ObjectProphecy;
+use Stripe\Collection;
+use Stripe\StripeObject;
+
+trait StripeFormattingTrait
+{
+    use ProphecyTrait;
+
+    protected function buildAutoIterableCollection(string $json): Collection|ObjectProphecy
+    {
+        /** @var \stdClass $itemsArray */
+        $itemsArray = json_decode($json, false);
+        /** @var StripeObject[] $itemData */
+        $itemData = $itemsArray->data;
+
+        $iterableCollection = $this->prophesize(Collection::class);
+        $iterableCollection->autoPagingIterator()->willReturn(new ArrayIterator($itemData));
+        $iterableCollection->count()->willReturn(count($itemData));
+
+        return $iterableCollection->reveal();
+    }
+}
```

Comment:
I'm not really a fan of traits - they couple to the class using them in a bit of an uncontrollable way. Since this doesn't access any properties of the class that uses it, except via `prophesize`, and prophesize is just to used to create essentially a value object, there's no real reason for it to be a trait. Afaik it would be a bit nice as a regular class with a static method.

Only trouble is it seems awkward to make the autoPagingIterator with prefilled data. There might be a nicer way to do it, but this is what I came up with:

```suggestion
<?php

namespace BigGive\Identity\Tests;

use ArrayIterator;
use Stripe\Collection;
use Stripe\StripeObject;

class StripeFormatting
{
    /**
     * @psalm-suppress MissingTemplateParam
     * @psalm-suppress InvalidReturnType
     * @psalm-suppress InvalidReturnStatement
     */
    public static function buildAutoIterableCollection(string $json): Collection
    {
        /** @var \stdClass $itemsArray */
        $itemsArray = json_decode($json, false);
        /** @var StripeObject[] $itemData */
        $itemData = $itemsArray->data;

        return new class ($itemData) extends Collection {
            /**
             * @param StripeObject[] $itemData
             */
            public function __construct(private array $itemData)
            {
                parent::__construct();
            }

            public function autoPagingIterator()
            {
                return new ArrayIterator($this->itemData);
            }
        };
    }
}
```